### PR TITLE
fix duplicate extension checks

### PR DESCRIPTION
**Pull request by moving-bits**
- fixes the error and suggestions mentioned in #8445
- adds the missing `.loc` file extension
- moves the known file extensions to `FileUtils`

### CODE REVIEW

File: main/src/cgeo/geocaching/HandleLocalFilesActivity.java [hunk 1]
```diff
@@ -29,10 +34,10 @@ public void onCreate(final Bundle savedInstanceState) {
                 final int startExtension = parsedUri.lastIndexOf('.');
                 if (startExtension > -1) {
                     final String extension = parsedUri.substring(startExtension).toLowerCase();
-                    if (".zip".equals(extension) || ".gpx".equals(extension) || ".gpx".equals(extension)) {
+                    if (ZIP_FILE_EXTENSION.equalsIgnoreCase(extension) || GPX_FILE_EXTENSION.equalsIgnoreCase(extension) || COMPRESSED_GPX_FILE_EXTENSION.equalsIgnoreCase(extension) || LOC_FILE_EXTENSION.equalsIgnoreCase(extension)) {
```
Comment (UniQP) [inline main/src/cgeo/geocaching/HandleLocalFilesActivity.java:37]:
```suggestion
                    if (extension.equals(ZIP_FILE_EXTENSION) || extension.equals(GPX_FILE_EXTENSION) || extension.equals(COMPRESSED_GPX_FILE_EXTENSION) || extension.equals(LOC_FILE_EXTENSION)) {
```

File: main/src/cgeo/geocaching/HandleLocalFilesActivity.java [hunk 2]
```diff
@@ -29,10 +34,10 @@ public void onCreate(final Bundle savedInstanceState) {
...
                         continueWith(CacheListActivity.class, intent);
                         finished = true;
-                    } else if (".map".equals(extension)) {
+                    } else if (MAP_FILE_EXTENSION.equalsIgnoreCase(extension)) {
```
Comment (UniQP) [inline main/src/cgeo/geocaching/HandleLocalFilesActivity.java:40]:
```suggestion
                    } else if (extension.equals(MAP_FILE_EXTENSION)) {
```

File: main/src/cgeo/geocaching/HandleLocalFilesActivity.java (hunk 1 above)
Comment (moving-bits) [reply main/src/cgeo/geocaching/HandleLocalFilesActivity.java:37]: In know this looks a bit awkward, but the other way around Codacy complains with "Issue found: Position literals first in String comparisons"

Comment (UniQP) [reply main/src/cgeo/geocaching/HandleLocalFilesActivity.java:37]: But we are using constants now, not literals.

Comment (moving-bits) [reply main/src/cgeo/geocaching/HandleLocalFilesActivity.java:37]: Ah, I see. Thanks for pointing that out to me. - I thought, Codacy would handle them the same. Have updated the PR now.

**Comment by moving-bits** [comment]
retest this please

**Comment by moving-bits** [comment]
retest this please

**Pull request merged**